scripts/zones/Dragons_Aery/mobs/Fafnir.lua

```diff
@@ -10,6 +10,7 @@ local entity = {}
 entity.onMobSpawn = function(mob)
     mob:setLocalVar('[rage]timer', 3600) -- 60 minutes
     mob:setMobMod(xi.mobMod.WEAPON_BONUS, 50) -- Level 90 + 50 = 140 Base Weapon Damage
+    mob:setMod(xi.mod.SLEEPRESBUILD, 3) -- decrease the duration of sleep spells 3 seconds for each successful sleep
```

Comment:
If I'm understanding this correctly, just from a quick glance, this system relies on mobs having a modifier in there already, in order for it to work?

Reply:
Its a good first effort, dont get me wrong, but current implementation would mean having to add a line of code, per status effect per mob script in order to fully introduce.

I think we need more data on how he system actually works in retail. Concretely:
- Which mobs build resistance? Is it a NM global thing or just certain mobs?
- Does the duration variation vary in 3 by 3 seconds in all cases or just in this concrete case and with some others is 5 or 2 (all regarding the same status effect, sleep in this case)?

Depending on this, the implementation could vary DRASTICALLY.

Reply:
> Its a good first effort, dont get me wrong, but current implementation would mean having to add a line of code, per status effect per mob script in order to fully introduce.
> 
> I think we need more data on how he system actually works in retail. Concretely:
> 
> * Which mobs build resistance? Is it a NM global thing or just certain mobs?
> * Is the duration decresed in sets of 3 seconds in all cases or just in this concrete case and with some others is 5 or 2 seconds (all regarding the same status effect, `sleep` in this case)?
> 
> Depending on this, the implementation could vary DRASTICALLY.

The reason for having the mod-based setup is because it appears to be limited to only a few NMs, currently we only know of Fafnir with Sleep. I will ask Jimmayus if they have more data.

Reply:
tbh we should probably not try to create the system before we have a fair sized list of places it will be used

Reply:
I've been asking around... Here is what I could gather (Take it with a BIG chunk of salt)
- Its a case by case system, where only certain mobs can build resistance against certain elements.
- The decrease seems to happen after the resistance check.
- It looks like a multiplicative decrease more like a flat second * uses reduction, similar to nuke wall. The decrease can floor duration to 1 second.
- The resistance resets at about 5 minutes of not landing the enfeeble

With all of this in mind:
- Its seems a lua edit/modifier placement per mob is inevitable. This could be alleviated via a unique modifier that acts as a bitmask for each effect or via sql.
- This would also mean that we could handle each resistance via editing the modifier directly OR use local variables and avoid defining 16+ new modifiers.

I will need to perform some reail testing and rumiate this over

Reply:
> This could be alleviated via a unique modifier that acts as a bitmask for each effect or via sql.

or localVar on the monster entity to track it

Reply:
> OR use local variables and avoid defining 16+ new modifiers.

hi I'm Teo and my hobbies include replying to Xaver before I have finished reading his posts 👋